### PR TITLE
Add geolocation capability to mobile route

### PR DESCRIPTION
**Pull request by parkershepherd**
Rather that just show the distance/direction from the server's scan origin, I added the ability for the mobile route to pass its location to the server. (Note: in Chrome, at least, you must browse using localhost, or else you must have https enabled)

## Description
Using the geolocation api, I query the devices location and pass it along to the server. Since https is required, I left this disabled by default, but configurable (via a checkbox/localStorage)

## Motivation and Context
Seeing the direction/distance from the "origin" and the pokemon is not nearly as useful as seeing the direction/distance from your current location.

## How Has This Been Tested?
Tested on Chrome/OSX, Safari/OSX, Firefox/OSX, and Chrome/Android
Browsed to `/mobile` and made sure it worked, made sure "Refresh" button worked normally, enabled "use device location" and refreshed again, noting that direction/distance information changed when the toggle was on.

The mobile page uses a separate endpoint, so this should not affect other features

## Screenshots (if appropriate):
screenshot at jul 23 17-17-08

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by andrewda** [comment]
Made a PR on your fork that you might want to pull in :)

File: pogom/app.py [hunk 1]
```diff
@@ -87,7 +87,12 @@ def next_loc(self):
     def list_pokemon(self):
         # todo: check if client is android/iOS/Desktop for geolink, currently only supports android
         pokemon_list = []
-        origin_point = LatLng.from_degrees(config['ORIGINAL_LATITUDE'], config['ORIGINAL_LONGITUDE'])
+
+        # Allow client to specify location
+        lat = request.args.get('lat', config['ORIGINAL_LATITUDE'], type=float)
```
Comment (machadox) [inline pogom/app.py]: Traceback (most recent call last):
  File "runserver.py", line 13, in <module>
    from pogom.app import Pogom
  File "/pokemon/pogom/app.py", line 97
    lat = request.args.get('lat', config['ORIGINAL_LATITUDE'], type=float)

**Comment by machadox** [comment]
im getting an error, but this would be awesome!!! gj

**Comment by parkershepherd** [comment]
@machadox was there more to that error message? I'm not sure what you're seeing

**Comment by adrianocr** [comment]
This doesn't include a map view? Only a list of pokemon in the area? Definitely useful on it's own but would be better with a map view IMO

**Comment by parkershepherd** [comment]
@adrianocr the main endpoint is still supported on mobile devices, this route was just created as a minimal-bandwidth alternative

File: static/dist/js/app.min.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
 /*
- pogomap 2016-07-23 
+ pogomap 2016-07-24
```
Comment (MonkeyLeeT) [inline static/dist/js/app.min.js]: Can we leave as it is?

**Comment by MonkeyLeeT** [comment]
@machadox Can you describe a bit more about your issue?

Otherwise this looks good to me. +1

**Comment by parkershepherd** [comment]
@MonkeyLeeT my mistake. I copied deploy's app.min.js over since this PR shouldn't have changed it

**Comment by parkershepherd** [comment]
@andrewda I made a commit based on your PR that just uses querySelector instead of jquery. What do you think?

**Comment by andrewda** [comment]
+1

**Comment by FabianSperrle** [comment]
Any chance of making that respect the current filters for hiding Pokémon as set in the "normal" map view?

**Pull request merged**